arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/typeclasses/Semigroup.kt

```diff
@@ -6,23 +6,27 @@ import arrow.core.Endo
 import arrow.core.Ior
 import arrow.core.NonEmptyList
 import arrow.core.Option
-import arrow.core.Validated
 import arrow.core.combine
 import arrow.core.compose
 import kotlin.jvm.JvmName
 import kotlin.jvm.JvmStatic
 
 public fun interface Semigroup<A> {
+  
+  // TODO: think of better name
+  public fun append(a: A, b: A): A
```

Comment:
is this still part of the Validated removal?
And is `append` going to replace `combine`

Reply:
I addressed it briefly in my comment, the problem is that the current `combine` signature doesn't match the signature we want for `zip`. So these changes belong together. There has been some talk about removing `Semigroup`, which would remove this changes (and horrible API name). And would replace the `Semigroup<E>` argument with a simple lambda `(E, E) -> E`. cc\\ @raulraja 

> Additional changes
>
> Semigroup
>
> To make this PR compile, and these functions to accept (E, E) -> E where fun interface Semigroup is defined the abstract function of Semigroup had to change shape from E.(E) -> E to (E, E) -> E. This is done in a non-binary breaking way, but we can still consider removing Semigroup in a different PR or changing its APIs.

Reply:
Sorry I missed that :)